gpm_api/tests/dataset/generate_test_granule_data.py

```diff
@@ -219,58 +251,71 @@ def create_test_hdf5(src_fpath, dst_fpath):
     dst_file.close()
 
 
-def cut_raw_assets():
-    filenames = os.listdir(raw_assets_dir_path)
+def cut_raw_granules():
+    product_basenames = os.listdir(granules_dir_path)
 
-    for filename in filenames:
-        print(f"Cutting {filename}")
-        raw_asset_filepath = os.path.join(raw_assets_dir_path, filename)
-        cut_asset_filepath = os.path.join(cut_assets_dir_path, filename)
+    for product_basename in product_basenames:
+        print(f"Cutting {product_basename}")
+        raw_filepath = os.path.join(
+            granules_dir_path, product_basename, RAW_DIRNAME, product_basename + ".HDF5"
+        )
+        cut_dir_path = os.path.join(granules_dir_path, product_basename, CUT_DIRNAME)
+        cut_filepath = os.path.join(cut_dir_path, product_basename + ".HDF5")
+        os.makedirs(cut_dir_path, exist_ok=True)
         try:
-            create_test_hdf5(raw_asset_filepath, cut_asset_filepath)
+            create_test_hdf5(raw_filepath, cut_filepath)
         except Exception as e:
-            print(f"Failed to cut {filename}: {e}")
+            print(f"Failed to cut {product_basename}: {e}")
 
 
-cut_raw_assets()
+cut_raw_granules()
 
 
-# Open assets with gpm_api and save as netCDF ##################################
+# Open granules with gpm_api and save as netCDF ################################
 
 
-def open_and_save_processed_assets(products: dict):
-    filenames = os.listdir(cut_assets_dir_path)
+def open_and_save_processed_granules(products: dict):
+    product_basenames = os.listdir(granules_dir_path)
 
     for product, product_info in products.items():
-        filename = find_filename_from_pattern(filenames, product_info["pattern"])
-        if filename is None:
+        if "start_time" not in product_info:
+            continue
+
+        product_basename = find_product_basename_from_pattern(
+            product_basenames, product_info["pattern"]
+        )
+        if product_basename is None:
             print(f"Could not find {product} file")
             continue
 
-        process_asset(filename, product_info["scan_modes_v7"])
+        version = gpm_products.get_last_product_version(product)
+        scan_modes = product_info["scan_modes"][f"V{version}"]
+        process_granule(product_basename, scan_modes)
 
 
-def find_filename_from_pattern(filenames: list[str], pattern: str) -> str | None:
-    for filename in filenames:
-        if pattern.rstrip("*").rstrip("\\d-") in filename:  # TODO: clarify specs of pattern
-            return filename
+def find_product_basename_from_pattern(product_basenames: list[str], pattern: str) -> str | None:
+    for product_basename in product_basenames:
+        if pattern.rstrip("*").rstrip("\\d-") in product_basename:  # TODO: clarify specs of pattern
```

Comment:
Not sure I understand what's going on here ... but let's discuss next week @sphamba